touchforms/formplayer/static/formplayer/script/webformsession.js

```diff
@@ -139,14 +140,33 @@ WebFormSession.prototype.serverRequest = function (requestParams, callback, bloc
 
     this.numPendingRequests++;
     this.onLoading();
-    $.ajax({
-            type: 'POST',
-            url: url,
-            data: JSON.stringify(requestParams),
-            dataType: "text"  // we don't use JSON because of a weird bug: http://manage.dimagi.com/default.asp?190983
-        })
-        .success(function(resp) { self.handleSuccess(JSON.parse(resp), callback); })
-        .fail(function (resp, textStatus) { self.handleFailure(JSON.parse(resp), textStatus); });
+
+    if(self.formplayerEnabled){
+        $.ajax({
+                type: 'POST',
+                url: 'http://localhost:8090/' + requestParams.action,
```

Comment:
also a fair bit of this can be moved after the else clause the success and fail functions are the same and many of the options are as well